### PR TITLE
Add multisignature class - Closes #886

### LINKED ISSUE #886: Add MultiSignatureTransaction class
### Description

Extend `BaseTransaction` implement all the specific features of multi-signature registration transactions.

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### Description

Added multisignature registration class.

### Review checklist

* The PR resolves #886
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transactions/src/transactions/4_register_multisignature.ts [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+					'Invalid multisignature keysgroup. Can not contain sender',
+					this.id,
+					'.signatures',
+				),
+			);
+		}
+
+		const updatedSender = {
+			...state.sender,
+			multisignatures: this.asset.multisignature.keysgroup.map(key =>
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/4_register_multisignature.ts:418]: I think `multimin` and `multilifetime` needs to be updated, too

File: packages/lisk-transactions/src/transactions/4_register_multisignature.ts [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+	}
+
+	public undo({ sender }: RequiredMultisignatureState): TransactionResponse {
+		const { errors: baseErrors, state } = super.undo({ sender });
+		if (!state) {
+			throw new Error('State is required for undoing transaction');
+		}
+		const errors = [...baseErrors];
+
+		const { multisignatures, ...strippedSender } = state.sender;
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/4_register_multisignature.ts]: same as above, but opposite

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 3]
```diff
@@ -29,6 +29,8 @@ import {
 	TransactionResponse,
 } from './base';
 
+const TRANSACTION_SIGNATURE_TYPE = 1;
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts:32]: maybe it's better to separate the issue to add type

File: packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts [hunk 4]
```diff
@@ -59,8 +61,10 @@ export const secondSignatureAssetTypeSchema = {
 					type: 'string',
 				},
 			},
+			additionalProperties: false,
```
Comment (shuse2) [inline packages/lisk-transactions/src/transactions/1_register_second_passphrase.ts]: same as above, it's better to separate the issue to add this check

**Review by shuse2 — DISMISSED** [review]

File: packages/lisk-transactions/src/transactions/4_register_multisignature.ts [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+		return {
+			id: this.id,
+			status: errors.length === 0 ? Status.OK : Status.FAIL,
+			errors,
+		};
+	}
+
+	public apply({ sender }: RequiredMultisignatureState): TransactionResponse {
+		const { errors: baseErrors, state } = super.apply({ sender });
```
Comment (SargeKhan) [inline packages/lisk-transactions/src/transactions/4_register_multisignature.ts:383]: I don't like the fact that in some transactions, we use the state returned by `super.apply` while in others, we don't. For example, in type 0 transaction, the state returned by `super.apply` is not used. 
What should be the consistent approach?

File: packages/lisk-transactions/test/transactions/4_register_multisignature.ts [hunk 6]
```diff
@@ -0,0 +1,627 @@
...
+	});
+
+	describe('#constructor', () => {
+		it('should create instance of MultisignatureTransaction', async () => {
+			expect(validTestTransaction)
+				.to.be.an('object')
+				.and.be.instanceof(MultisignatureTransaction);
+		});
+
+		it('should set multisignature asset', () => {
```
Comment (SargeKhan) [inline packages/lisk-transactions/test/transactions/4_register_multisignature.ts]: some tests are not async

File: packages/lisk-transactions/src/transactions/4_register_multisignature.ts (hunk 5 above)
Comment (shuse2) [reply packages/lisk-transactions/src/transactions/4_register_multisignature.ts:383]: let's create anther issue and consistently use state from the base transaction.

**Review by SargeKhan — DISMISSED** [review]
Made some comments for requested changes.

**Review by shuse2 — APPROVED** [review]

File: packages/lisk-transactions/test/transactions/4_register_multisignature.ts (hunk 6 above)
Comment (SargeKhan) [reply packages/lisk-transactions/test/transactions/4_register_multisignature.ts]: This test should also be asyc.

**Review by SargeKhan — APPROVED** [review]

**Pull request merged**